provider/maas/maas2instance.go

```diff
@@ -65,10 +71,19 @@ func (mi *maas2Instance) Addresses() ([]network.Address, error) {
 // Status returns a juju status based on the maas instance returned
 // status message.
 func (mi *maas2Instance) Status() instance.InstanceStatus {
-	// TODO (babbageclunk): this should rerequest to get live status.
-	statusName := mi.machine.StatusName()
-	statusMsg := mi.machine.StatusMessage()
+	args := gomaasapi.MachinesArgs{SystemIDs: []string{mi.machine.SystemID()}}
+	machines, err := mi.maasController.Machines(args)
+	if err != nil {
```

Comment:
Why are we not just returning the error? It should be up to the instance poller to make a call about what it does with the error. This is not a MAAS-specific issue.